packages/mobile/src/components/audio/Audio.tsx

```diff
@@ -58,6 +58,14 @@ const styles = StyleSheet.create({
   }
 })
 
+const getStreamUris = (track: Track, trackOwner: UserMetadata) => {
+  const encodedTrackId = encodeHashId(track.track_id)
+  const creatorNodeEndpoints = trackOwner.creator_node_endpoint.split(',')
+  return creatorNodeEndpoints.map(
```

Comment:
so is it the case that hls would figure out which endpoints were the best, wheras now we are just kinda guessing? assuming this isn't a big deal, but thought id ask so i understand any potential tradeoffs

Reply:
So I don't know the HLS stuff very well, but I think we were giving it the choice of endpoints to grab each segment. Before, we passed all the gateways as options which the player can use as it pleases. Presumably, it tries each one.
```
const gateways = audiusBackendInstance.getCreatorNodeIPFSGateways(
  trackOwner.creator_node_endpoint
)

const m3u8 = hlsUtils.generateM3U8Variants({
  segments: track.track_segments,
  gateways
})
```
The new logic will simply try each endpoint and fall back to the next if it errors out.
See `onError` in the [video player usage example](https://github.com/react-native-video/react-native-video/blob/master/API.md#usage)

Reply:
Thanks for context